### DISCUSSION TITLE
Does Serverpod support localization for email and web templates used for authentication, verrification, resets etc

### DISCUSSION THREAD

**Discussion by oscargoldman** [serverpod/serverpod#3032] [General]
I'm working on a project that is designated for international release. The client localization is easy enough, but the few server technologies I've looked at don't seem to easily support localization of things like emails for email verification, password reset etc, and the corresponding web pages for such features. Is Serverpod set up to easily handle this?

**Comment by SandPod** [comment]
Hey @oscargoldman!

The authentication module provided with Serverpod unfortunately does not provide great support for localization. There is nothing in the framework that present better localization support, but the implementation in the module does not natively provide that information in an accessible way.

### Possible workaround
It is pretty simple to store additional information connected to the user information in a table. And when you want to send an email, or provide other information that should be localized, you can do a lookup in that table to figure out what language the message should be transmitted in.

The user information is provided in all required environments so it is pretty simple to make a query based on that information.

And if performance is a concern, simply use the build in cache mechaninsm to store that information with a reasonable timeout when it has been retrieved.

### Future plans
We would like to introduce a "call context" object in the future where additional information could be passed on all or some requests from the client to the server. This would greatly simplify localization.
